addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -928,39 +928,59 @@ var SnippetEditor = Widget.extend({
         this.options.wysiwyg.odooEditor.automaticStepUnactive();
         var self = this;
 
+        // Snippets for which the grid mode should not be toggled on drag.
+        const untoggleableColumns = '.s_masonry_block, .s_showcase, .s_features_grid, .s_website_form, .s_media_list, .s_table_of_content, .s_process_steps';
+        const disableToggle = self.$target[0].closest(untoggleableColumns);
```

Comment:
I could use that directly if there was not the special case of Masonry: the option excludes this snippet because we don't need the option to toggle because it will always be in grid mode. But since it is in grid mode, it should still be "toggleable" in `_dragAndDropStart` given the way I made the condition (it just won't toggle for real because I check if the `o_grid_mode` class is there before doing so). Moreover, it's not visible here because I added it in a later commit but now it is `.s_masonry_block:not([data-vxml])` to support the old version of Masonry (so it's not ugly when we drag it).

So instead, I defined a list in the website part (as you suggested above)

Reply:
> Moreover, it's not visible here because I added it in a later commit but now it is .s_masonry_block:not([data-vxml]) to support the old version of Masonry

That's fine, but in general: we don't support using options on snippet of older data-XXX version. The only purpose of the data-XXX system is to keep the snippets working *outside of edit mode* and indicate in edit mode those are outdated snippets (if someone drag and see something weird, it would be ok as you advertise in edit mode this is an outdated snippet that should be replaced).

> I could use that directly if there was not the special case of Masonry

Mmh, can't this be "if the grid option is displayed in the editor panel **or if the snippet is already in grid mode**" ? :thinking:

Reply:
Keep in mind that the community can create snippets to extend Odoo. That's why most of our UI definition is in XML the way it is. Having to fill a javascript list in a random code the community is not supposed to care about is not dev-friendly. Everything should be automatic when possible :+1: